File: src/tasks/model/impl/TaskData.java
package tasks.model.impl;

import static java.util.Objects.requireNonNull;

import java.util.Objects;
import tasks.model.Task;

final class TaskData {

  private final String label;
  private final Task.Status status;

  TaskData(String label, Task.Status status) {
    this.label = requireNonNull(label);
    this.status = requireNonNull(status);
  }

  String label() {
    return label;
  }

  Task.Status status() {
    return status;
  }

  @Override
  public boolean equals(Object other) {
    return other instanceof TaskData
        && Objects.equals(this.label, ((TaskData) other).label)
        && Objects.equals(this.status, ((TaskData) other).status);
  }

  @Override
  public int hashCode() {
    return Objects.hash(label, status);
  }
}

File: src/tasks/model/impl/TaskData.kt
package tasks.model.impl

import java.util.Objects
import tasks.model.Task

class TaskData(private val label: String, private val status: Task.Status) {
  fun label(): String {
    return label
  }

  fun status(): Task.Status {
    return status
  }

  override fun equals(other: Any?): Boolean {
    return other is TaskData && label == other.label && status == other.status
  }

  override fun hashCode(): Int {
    return Objects.hash(label, status)
  }

}